src/Libraries/Nop.Core/Caching/MemoryCacheManager.cs

```diff
@@ -117,15 +107,16 @@ public async Task<T> GetAsync<T>(CacheKey key, Func<Task<T>> acquire)
             if ((key?.CacheTime ?? 0) <= 0)
                 return await acquire();
 
-            if (_memoryCache.TryGetValue(key.Key, out T result))
-                return result;
-
-            result = await acquire();
-
-            if (result != null)
-                await SetAsync(key, result);
-
-            return result;
+            var value = await _memoryCache.GetOrCreate(
+                key.Key,
+                entry =>
+                {
+                    entry.SetOptions(PrepareEntryOptions(key));
+                    return new Lazy<Task<T>>(acquire, true);
+                }).Value;
+            if (value == null)
+                await RemoveAsync(key);

```

Comment:
This is problematic - because the fact that caching null is disallowed is (ab-)used by services to avoid setting values, RedisSynchronizedMemoryCache unnecessarily spams update events for removing null values. I think a better pattern would be to allow and guard against the possibility of null being cached, and where necessary pass `acquire = null` (instead of `acquire = () => null`.

Reply:
Resolved in commit [1570d47](https://github.com/nopSolutions/nopCommerce/pull/6531/commits/1570d47bbb26fca389f7b3416327317b82452d52)